File: golang/speedtest.go
package main
import ("fmt"; "time")

// The average execution time of this script is 51.44 ms after 10 test

const ITERATIONS int = 10;
const NUMBER int = 100000000;

func speedtest() time.Duration {
	var start time.Time = time.Now();

	var count int = 0;
	for i := 0; i < NUMBER; i++ {
		count++;
	}

	var elapsed time.Duration = time.Since(start);
	return elapsed;
}

func main() {
	var totalExecutionTime, finalDuration time.Duration;

	fmt.Printf("Starting speed test...\n")

	for i := 0; i < ITERATIONS; i++ {
		finalDuration = speedtest();
		totalExecutionTime += finalDuration;
		fmt.Printf("%d. Counting to number %d in Golang took %v\n", i + 1, NUMBER, finalDuration);
	}

	averageExecutionTime := totalExecutionTime / time.Duration(ITERATIONS)
	fmt.Printf("\n\033[0;32mAverage execution time: %v\n", averageExecutionTime)
}

File: rust/speedtest.rs
use std::time::{Duration, Instant};

// The average execution time of this script is 578.50 ms after 10 test

const ITERATIONS: usize = 10;
const NUMBER: usize = 100_000_000;

fn speedtest() -> Duration {
    let start = Instant::now();

    let mut count = 0;
    for _ in 0..NUMBER {
        count += 1;
    }

    start.elapsed()
}

fn main() {
    let mut total_execution_time = Duration::new(0, 0);

    println!("Starting speed test...");

    for i in 0..ITERATIONS {
        let final_duration = speedtest();
        total_execution_time += final_duration;
        println!("{}. Counting to number {} in Rust took {:?}", i + 1, NUMBER, final_duration);
    }

    let average_execution_time = total_execution_time / ITERATIONS as u32;
    println!("\n\u{1b}[0;32mAverage execution time: {:?}", average_execution_time);
}
